src/uucore/src/lib/features/quoting_style.rs

```diff
@@ -341,6 +342,63 @@ pub fn escape_name(name: &OsStr, style: &QuotingStyle) -> String {
     }
 }
 
+pub fn escape_dir_name(dir_name: &OsStr, style: &QuotingStyle) -> String {
+    match style {
+        QuotingStyle::Literal { show_control } => {
+            if *show_control {
+                dir_name.to_string_lossy().into_owned()
+            } else {
+                dir_name
+                    .to_string_lossy()
+                    .chars()
+                    .flat_map(|c| EscapedChar::new_literal(c).hide_control())
+                    .collect()
```

Comment:
At first glance, these code coverage warnings seem correct, but Codecov is very much known for its false positives.

Can you point out which test was supposed to set `show_control = false`? If there is no such test, please add a test for the code, to demonstrate that this is how and when it's needed.

Reply:
For now, I only copy-pasted the `escape_name` function and made small changes to adapt, but I still have to simplify and factor it.

Reply:
> copy-pasted

Code-reuse is fine, but code duplication is a red flag, usually implying that there is a better pattern that would more accurately describe the behavior. Also, code duplication means that whenever something changes, *someone* has to remember that it needs to be changed in two placed in the same way. This leads to so many problems. Please don't do that.

Yes, please go ahead and consider why it has to behave like this, what's the pattern behind it, and factor the code accordingly.

Reply:
Sure, this isn't meant to be permanent, but I am not sure of the pattern yet.

Plus, I found an escaping issue that has nothing to do with directory names printing, and I am going down this rabbit hole as well.